components/Layout.js

```diff
@@ -3,7 +3,12 @@ import Head from "next/head";
 import Header from "./Header";
 import Footer from "./Footer";
 
-export default function Layout({ children }) {
+export default function Layout({
+  children,
+  activePage,
+  setActivePage,
```

Comment:
As far as I understand it you do not need to recive setActivePage here because you already recive the toggleActivePage function, which does the same. Same with Header and Footer component.